packages/network/test/unit/ServerWsEndpoint.test.ts

```diff
@@ -22,39 +29,46 @@ describe('ServerWsEndpoint', () => {
         )
         serverWsEndpoint = new ServerWsEndpoint('127.0.0.1', wssPort1, false, httpServer, PeerInfo.newTracker('tracker'))
 
-        const ws = new WebSocket(serverWsEndpoint.getUrl() + '/ws',
-            undefined, {
-                headers: {
-                    'streamr-peer-id': 'peerId'
-                }
-            })
-        ws.once('error', (err) => {
-            throw err
-        })
-        await waitForEvent(ws, 'open')
-        ws.close()
+        const webSocketClient = new w3cwebsocket(
+            serverWsEndpoint.getUrl() + '/ws'
+        )
+        webSocketClient.onmessage = (message) => {
+            const { uuid, peerId } = JSON.parse(message.data.toString())
+            if (uuid && peerId) {
+                webSocketClient.send(JSON.stringify({uuid, peerId: 'peerId'}))
+                webSocketClient.close()
+            }
+        }
+        webSocketClient.onerror = (error) => { throw error }
+        await waitForCondition(() => webSocketClient.readyState === webSocketClient.CLOSED)
     })
 
     test('receives encrypted connections', async () => {
+        if (typeof _streamr_electron_test !== 'undefined') {
+            return
+        }
         const httpsServer = await startHttpServer(
             '127.0.0.1',
             wssPort2,
             'test/fixtures/key.pem',
             'test/fixtures/cert.pem'
         )
         serverWsEndpoint = new ServerWsEndpoint('127.0.0.1', wssPort2, true, httpsServer, PeerInfo.newTracker('tracker'))
-
-        const ws = new WebSocket(serverWsEndpoint.getUrl() + '/ws',
-            undefined, {
-                rejectUnauthorized: false, // needed to accept self-signed certificate
-                headers: {
-                    'streamr-peer-id': 'peerId'
-                }
-            })
-        ws.once('error', (err) => {
-            throw err
-        })
-        await waitForEvent(ws, 'open')
-        ws.close()
+        const webSocketClient = new w3cwebsocket(
+            serverWsEndpoint.getUrl() + '/ws',
+            undefined,
+            undefined,
+            undefined,
+            { rejectUnauthorized: false }
+        )
+        webSocketClient.onmessage = (message) => {
+            const { uuid, peerId } = JSON.parse(message.data.toString())
+            if (uuid && peerId) {
+                webSocketClient.send(JSON.stringify({uuid, peerId: 'peerId'}))
+                webSocketClient.close()
+            }
+        }
+        webSocketClient.onerror = (error) => { throw error }
+        await waitForCondition(() => webSocketClient.readyState === webSocketClient.CLOSED)
```

Comment:
There should be an assertion in this test that the connection got opened at some point?